tools/setup-envtest/main.go

```diff
@@ -0,0 +1,292 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2021 The Kubernetes Authors
+
+package main
+
+import (
+	goflag "flag"
+	"fmt"
+	"os"
+	"path/filepath"
+	"runtime"
+
+	"github.com/go-logr/logr"
+	"github.com/go-logr/zapr"
+	flag "github.com/spf13/pflag"
+	"go.uber.org/zap"
+	"github.com/spf13/afero"
+
+	envp "sigs.k8s.io/controller-runtime/tools/setup-envtest/env"
+	"sigs.k8s.io/controller-runtime/tools/setup-envtest/remote"
+	"sigs.k8s.io/controller-runtime/tools/setup-envtest/versions"
+	"sigs.k8s.io/controller-runtime/tools/setup-envtest/workflows"
+)
+
+
+const (
+	// envNoDownload is an env variable that can be set to always force
+	// the --installed-only, -i flag to be set.
+	envNoDownload = "ENVTEST_INSTALLED_ONLY"
```

Comment:
Thoughts on `CONTROLLER_RUNTIME_ENVTEST_` as a prefix instead of `ENVTEST_`? Eventually it would be nice to add similarly prefixed variables to the set for specifying envtest config options, so it would also be nice to have all envtest-related config variables aligned in name. I guess the tool's config != envtest config, so maybe `CONTROLLER_RUNTIME_ENVTEST_TOOL_`.

Corrollary: remove variables prefixed with `KUBEBUILDER_` prior to 1.0 since envtest is more general than kubebuilder.

Reply:
I'm kinda :woman_shrugging: on variable names, except that I don't like holding down the shift key much :-P.

Seriously though, maybe just ENVTEST_TOOL_XYZ?